RedHatInsights/ccx-notification-service
# Performance optimizitations round 1

# Description

This service is taking way more time than expected when dealing with our production data. These "small" optimisations aim to improve its performance  by reducing to a minimum the amount of work done. The idea is to decide if there is a need to introduce concurrency  AFTER verifying the improvements in production.

## Type of change

- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- Refactor (refactoring code, removing useless files)

## Testing steps

- Updated existing UTs and validated new behaviour through previously tested scenarios 
- `make bdd_tests` passes
- Each commit of this PR has been tested and benchmarked separately to see if it optimises the processing or not.

## Checklist
